File: JS/sort/lowOrderPrint.js
"use strict";

const fs = require("fs");

const input =
  process.platform === "linux"
    ? fs.readFileSync("/dev/stdin").toString().split("\n")
    : `2
홍길동 95
이순신 77`.split("\n");

const N = input[0];

const students = input.splice(1).map((item) => ({
  name: item.split(" ")[0],
  score: Number(item.split(" ")[1]),
}));

students.sort((a, b) => a.score - b.score);

students.forEach((item) => console.log(item.name));

File: src/sort/LowOrderPrint.java
package sort;

import java.io.*;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class LowOrderPrint {
    static class Student implements Comparable<Student>{
        String name;
        int score;

        public Student(String name, int score) {
            this.name = name;
            this.score = score;
        }

        public String getName() {
            return name;
        }

        public int getScore() {
            return score;
        }

        @Override
        public int compareTo(Student o) {
            if(score < o.getScore()) {
                return -1;
            }else if(score > o.getScore()){
                return 1;
            }

            return 0;
        }
    }

    public static void main(String[] args) throws IOException {
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(br.readLine());

        List<Student> list = new ArrayList<>(N);

        for (int i = 0; i < N; i++) {
            String[] in = br.readLine().split(" ");

            list.add(i, new Student(in[0], Integer.parseInt(in[1])));
        }

        Collections.sort(list);

        StringBuilder sb = new StringBuilder();

        list.forEach(std -> sb.append(std.getName() + " "));

        bw.write(sb.toString());
        bw.flush();
        bw.close();
    }
}
